### ISSUE TITLE
Support viewing different Git Branches. Currently, viewing source from other git tree/branch is broken.

### ISSUE THREAD

**Issue by jiayounokim** [conwnet/github1s#22]
Try going to https://github1s.com/JetBrains/kotlin/tree/1.0.1_/br145

The files are not displayed!

Although files are displayed as expected for Master branch here:  https://github1s.com/JetBrains/kotlin

This could mean different git tree/branch is not supported and files are not shown.

**Comment by conwnet** [comment]
It seems the branch contains the character `/` and broken the url parser, may be we have to fetch all branches first for this repository and then judge which is the correct branch name, let me see...

for another branch such as 1.0.1 it works ok...[https://github1s.com/JetBrains/kotlin/tree/1.0.1](https://github1s.com/JetBrains/kotlin/tree/1.0.1)

**Comment by Kryptos-FR** [comment]
Anything after `tree/` would be the branch name, isn't it?

It's quite common to have branches containing the `/` separator, for instance when using git flow you have a lot of `feature/TICKET-ID`. And some projects might have other conventions as well.

**Comment by conwnet** [comment]
> Anything after `tree/` would be the branch name, isn't it?

Not all, when you enter any directories or open file, the file path will appends to the branch, and the separator of branch and filePath also is character `/`.

Infact, when your url has a filePath, github1s will auto open the corresponding file or focus corresponding directory.

**Comment by justanotheranonymoususer** [comment]
Hmm, so there's an ambiguity

> for example: https://github.com/conwnet/github1s/tree/master/src/vs

What if you also create a branch named `master/src/vs`? =)

**Comment by BigsonLvrocha** [comment]
I've created and issue, but closed it since I saw this one, there, I propose a solution

> The problem is in  [this line](https://github.com/conwnet/github1s/blob/820b09f853edd0e04c8cd39f382a62da55160dac/src/vs/github1s/util.ts#L26), maybe it should accept an argument for number of tries, when the api returns "not found" it gets incremented

but with @justanotheranonymoususer the branch master/src/vs would still be unacessible, it's and edge case, but also one that is a case that happens when good github practices are not followed

**Comment by BigsonLvrocha** [comment]
each "not found" the number gets incremented and tries to find a branch name using more values in path array

**Comment by Hipska** [comment]
I was expecting to be able to use the "Source Control" in order to switch branches, but that does not seem to work as well..

**Comment by xcv58** [comment]
We can use GraphQL query to get all branch name and use it to determine which parts of the URL is the valid branch:

https://docs.github.com/en/graphql/overview/explorer

```
query objectQuery($owner: String!, $repo: String!) {
  repository(name: $repo, owner: $owner) {
    id
    defaultBranchRef {
      name
      prefix
    }
    refs(refPrefix: "refs/heads/" first: 100) {
      totalCount
      nodes {
        name
      }
    }
  }
}
```

```json
{
  "data": {
    "repository": {
      "id": "MDEwOlJlcG9zaXRvcnkxOTIxNzUwOTU=",
      "defaultBranchRef": {
        "name": "master",
        "prefix": "refs/heads/"
      },
      "refs": {
        "totalCount": 2,
        "nodes": [
          {
            "name": "gh-pages"
          },
          {
            "name": "master"
          }
        ]
      }
    }
  }
}
```
![image](https://user-images.githubusercontent.com/503123/107834849-53743980-6d4c-11eb-810f-3e42fe34f194.png)

**Comment by xcv58** [comment]
I'm working on this, there are some bugs https://github.com/xcv58/github1s/pull/4

**Comment by softmarshmallow** [comment]
With surf command you can view other branches

``` sh
npm i --g @bridged.xyz/surf
surf . -b <branch-name>
```

this opens up your branch code on browser.

source code here [(cli)](https://github.com/bridgedxyz/surf-cli) and here [(app)](https://github.com/bridgedxyz/github.surf)

**Comment by lonelykid** [comment]
>for example: https://github.com/conwnet/github1s/tree/master/src/vs 
>Infact, when your url has a filePath, github1s will auto open the corresponding file or focus corresponding directory.

This is not working. for example: https://github1s.com/conwnet/github1s/tree/master/src/vs  is opened to the vs folder.

**Comment by conwnet** [comment]
> > for example: https://github.com/conwnet/github1s/tree/master/src/vs
> > Infact, when your url has a filePath, github1s will auto open the corresponding file or focus corresponding directory.
> 
> This is not working. for example: https://github1s.com/conwnet/github1s/tree/master/src/vs is not opened to the vs folder.

it seems a bug, we will fix it these days

**Comment by conwnet** [comment]
fixed in #167
